src/app/core/tools/handlers/SelectionHandler.ts

```diff
@@ -39,8 +39,7 @@ export const SelectionHandler: EventHandler = ({
             const deselect = (input.isShiftKeyDown() && selections.has(obj));
             action.add(new SelectAction(selections, obj, deselect).execute());
             action.add(new ShiftAction(designer, obj).execute());
+            history.add(action);
```

Comment:
i have a sneaking suspicion this is going to break deselect-ing, we do a `action.add` above that

please confirm

Reply:
should be fixed